strabo/private_helper.py

```diff
@@ -24,14 +24,18 @@ def fill_interest_point(ip,image_ids,form_title,form_body,form_geo_obj,form_laye
     ip.icon = form_icon
     ip.images = [db.session.query(schema.Images).get(int(id)) for id in image_ids]
 
-def fill_image(image,form_file_obj,form_descrip,year,month,day):
+def fill_image(image,form_file_obj,form_descrip,year,month):
     '''
-    if a flask.files object is passed in, then  
+    If a flask.files object is passed in, then it saves the new file and stores the new information in the database.
+    If the image row already stored infromation about an old image (the image was edited), then that old image is deleted.
     '''
-    image.taken_at = datetime.date(utils.safe_pos_int_conv(year),utils.safe_pos_int_conv(month),utils.safe_pos_int_conv(day))
+    image.taken_at = datetime.date(utils.safe_pos_int_conv(year),utils.safe_pos_int_conv(month),1)
     image.description = form_descrip
 
     if form_file_obj:
```

Comment:
form_file is much better. I guess I wanted to differentiate it from the other form arguments, which are all strings, but it is probably clear enough on its own. It is also unbelievably bad Hungarian notation.

For name abbreviations, there is an entire literature on how and when and how much to abbreviate, and I am definitely not an expert, but basically you can think of it as 3 different versions of the same word

* description (11 characters)
* descrip (7 characters)
* desc (4 characters)

I don't think I do this in the code, but potentially, I might just type "d" (1 character)

Fewer characters means less clear and more ambiguous, but shorter, which is easier to type and sometimes easier to read. 

I won't claim that this is a terribly good example, and I often am inconsistent for no good reason, but I do think it important in general to have multiple abbreviations for certain words or phrases. In the strabo code, I tried to lean towards really long names for maximum clarity, and if it is confusing, then we can change desc to descrip, but I think you may often find it rather verbose as you get more experience.